### ISSUE TITLE
model.coef_ load coeffients in wrong axis when using booster:gblinear and objective: multi:softprob

### ISSUE THREAD

**Issue by Bolozano** [dmlc/xgboost#9547]
Solution already found & How to reproduce:
```import json

import numpy as np
import xgboost as xgb

from scipy.special import expit
from sklearn.datasets import load_digits
from sklearn.model_selection import train_test_split

X, y = load_digits(n_class=3, return_X_y= True)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

model = xgb.XGBClassifier(booster='gblinear', learning_rate=0.1,
                               objective = 'multi:softprob', num_classes=3, n_jobs=1,verbosity=1)
model.fit(X_train, y_train)
print('predict_proba: ', model.predict_proba(X_test)[0])
coef = model.coef_
exp_output = np.exp(np.dot(coef, X_test.T).T + model.intercept_)
sft_max_output = exp_output/np.sum(exp_output, axis=1).reshape([-1, 1])
print('Using default model.coef_' ,sft_max_output[0])

model_dump = model.get_booster().get_dump(dump_format="json")
coef = np.array(json.loads(model_dump[0])["weight"], dtype=np.float32).reshape(64, -1).T
exp_output = np.exp(np.dot(coef, X_test.T).T + model.intercept_)
sft_max_output = exp_output/np.sum(exp_output, axis=1).reshape([-1, 1])
print('reshape coef follwing different axis: ' ,sft_max_output[0])
```

Normally you will get 

```[16:32:51] WARNING: ../src/learner.cc:767: 
Parameters: { "num_classes" } are not used.

predict_proba:  [1.4275121e-10 9.9986684e-01 1.3314626e-04]
Using default model.coef_ [1.00000000e+00 9.45980584e-13 8.11060201e-32]
reshape coef follwing different axis:  [1.42750991e-10 9.99866852e-01 1.33147465e-04]
```
By checking the source code of model.coef_ I found there is somehow a check for this but is not triggered, I am wondering if there is anything wrong in my case.
```
@property
    def coef_(self) -> np.ndarray:
        """
        Coefficients property

        .. note:: Coefficients are defined only for linear learners

            Coefficients are only defined when the linear model is chosen as
            base learner (`booster=gblinear`). It is not defined for other base
            learner types, such as tree learners (`booster=gbtree`).

        Returns
        -------
        coef_ : array of shape ``[n_features]`` or ``[n_classes, n_features]``
        """
        if self.get_xgb_params()["booster"] != "gblinear":
            raise AttributeError(
                f"Coefficients are not defined for Booster type {self.booster}"
            )
        b = self.get_booster()
        coef = np.array(json.loads(b.get_dump(dump_format="json")[0])["weight"])
        # Logic for multiclass classification
        n_classes = getattr(self, "n_classes_", None)
        if n_classes is not None:
            if n_classes > 2:
                assert len(coef.shape) == 1
                assert coef.shape[0] % n_classes == 0
                coef = coef.reshape((n_classes, -1))
        return coef
```
